core/src/main/java/de/danoeh/antennapod/core/util/PodcastIndexTranscriptUtils.java

```diff
@@ -51,4 +56,60 @@ public static void storeTranscript(FeedMedia media, String transcript) {
             e.printStackTrace();
         }
     }
+
+    public static Transcript loadTranscript(FeedMedia media) {
+        String type = media.getItem().getPodcastIndexTranscriptType();
+        String url = media.getItem().getPodcastIndexTranscriptUrl();
+        if (url == null) {
+            return null;
+        }
+
+        if (media.getItem().getTranscript() != null) {
+            return media.getItem().getTranscript();
+        }
+
+        if (media.getItem().getTranscript() != null) {
+            return media.getItem().getTranscript();
+        }
+
+        if (media.getTranscriptFile_url() != null) {
+            File dest = new File(media.getTranscriptFile_url());
+            if (dest.exists()) {
+                String transcriptStr = "";
+                try {
+                    BufferedReader reader = new BufferedReader(new FileReader(media.getTranscriptFile_url()));
+                    StringBuilder stringBuilder = new StringBuilder();
+                    char[] buffer = new char[128];
+                    while (reader.read(buffer) != -1) {
+                        stringBuilder.append(new String(buffer));
+                        buffer = new char[128];
+                    }
+                    reader.close();
+                    transcriptStr = stringBuilder.toString();
+                } catch (FileNotFoundException e) {
+                    e.printStackTrace();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
```

Comment:
I would move that to something like `PodcastIndexTranscriptUtils.loadTranscriptFromFile`. Also, for loading a file to a String, you can use the commons-io library with `FileUtils.readFileToString()`, which should be a lot easier to read